runtime/src/commonMain/kotlin/app/cash/sqldelight/Query.kt

```diff
@@ -25,13 +26,13 @@ import app.cash.sqldelight.db.use
  * @param RowType the type that this query can map its result set to.
  */
 @Suppress("FunctionName") // Emulating a constructor.
-fun <RowType : Any> Query(
+fun <RowType : Any, StatementType : SqlPreparedStatement, CursorType : SqlCursor> Query(
```

Comment:
I think you don't need `StatementType` for any of this and can just use `SqlPreparedStatement` in the `SqlDriver` api wherever its needed, the statement type only matters in the subclasses of query and not in this file.

Reply:
Do you envision us having something like `PgQuery`?

Reply:
no

Reply:
I see what you're saying now. Resolved.